server/server.go

```diff
@@ -521,6 +538,102 @@ func (s *server) Stop(ctx context.Context, req *rpcpb.StopRequest) (*rpcpb.StopR
 	return &rpcpb.StopResponse{ClusterInfo: info}, nil
 }
 
+func (s *server) AttachPeer(ctx context.Context, req *rpcpb.AttachPeerRequest) (*rpcpb.AttachPeerResponse, error) {
+	zap.L().Debug("received attach peer request")
+	info := s.getClusterInfo()
+	if info == nil {
+		return nil, ErrNotBootstrapped
+	}
+
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	nodes, err := s.network.nw.GetAllNodes()
+	if err != nil {
+		return nil, err
+	}
+	node, ok := nodes[req.NodeName]
+	if !ok {
+		return nil, ErrNodeNotFound
+	}
```

Comment:
If we're just getting 1 node by its name we can do `s.network.nw.GetNode(req.NodeName)` instead